include/RAJA/LegacyCompatibility.hxx

```diff
@@ -0,0 +1,215 @@
+#ifndef KRIPKE_LEGAGY_COMPATIBILITY_HXX
+#define KRIPKE_LEGAGY_COMPATIBILITY_HXX 1
+
+static_assert(__cplusplus >= 201103L, "C++ standards below 2011 are not supported");
+
+#include <type_traits>
+#include <utility>
+#include <iostream>
+#include <functional>
+#include <cstdint>
+
+namespace VarOps {
```

Comment:
Minor nitpick: `RAJA::internal` isn't very descriptive.  But for internal namespaces you've got some leeway because you can change them later.